2011-02-17  Andreas Schwab  <schwab@redhat.com>

	[BZ #12454]
	* elf/dl-deps.c (_dl_map_object_deps): Signal error early when
	dependencies are missing.
 
---
 elf/dl-deps.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/elf/dl-deps.c
+++ b/elf/dl-deps.c
@@ -491,6 +491,10 @@
   if (errno == 0 && errno_saved != 0)
     __set_errno (errno_saved);
 
+  if (errno_reason)
+    _dl_signal_error (errno_reason == -1 ? 0 : errno_reason, objname,
+		      NULL, errstring);
+
   struct link_map **old_l_initfini = NULL;
   if (map->l_initfini != NULL && map->l_type == lt_loaded)
     {
@@ -683,8 +687,4 @@
     }
   if (old_l_initfini != NULL)
     _dl_scope_free (old_l_initfini);
-
-  if (errno_reason)
-    _dl_signal_error (errno_reason == -1 ? 0 : errno_reason, objname,
-		      NULL, errstring);
 }
